nodes/node_registry.go

```diff
@@ -63,31 +63,106 @@ func (r *NodeRegistry) GetRegisteredNodeKindNames() []string {
 	return result
 }
 
+func (r *NodeRegistry) GetGenerateNodeAttributes() map[string]*GenerateNodeAttributes {
+	result := map[string]*GenerateNodeAttributes{}
+	for k, v := range r.nodeIndex {
+		result[k] = v.GetGenerateAttributes()
+	}
+	return result
+}
+
 func (r *NodeRegistry) Kind(kind string) *NodeRegistryEntry {
 	return r.nodeIndex[kind]
 }
 
 type NodeRegistryEntry struct {
 	nodeKindNames []string
 	initFunction  Initializer
-	credentials   *Credentials
+	attributes    *NodeRegistryEntryAttributes
+}
+
+func (nre *NodeRegistryEntry) GetCredentials() *Credentials {
+	if nre.attributes == nil {
+		return nil
+	}
+	return nre.attributes.credentials
+}
+
+func (nre *NodeRegistryEntry) GetGenerateAttributes() *GenerateNodeAttributes {
+	if nre.attributes == nil {
+		return nil
+	}
+	return nre.attributes.generateAttributes
 }
 
 // Credentials returns entry's credentials.
-func (e *NodeRegistryEntry) Credentials() *Credentials {
+// might return nil if no default credentials present.
+func (e *NodeRegistryEntryAttributes) Credentials() *Credentials {
```

Comment:
looks like a dup of `func (nrea *NodeRegistryEntryAttributes) GetCredentials() *Credentials`

Reply:
038d326